includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -2079,24 +2089,25 @@ public function capture_charge( $order, $include_level3 = true ) {
 		$currency                 = WC_Payments_Utils::get_order_intent_currency( $order );
 
 		try {
-			$intent_id    = $order->get_transaction_id();
-			$intent       = $this->payments_api_client->get_intent( $intent_id );
+			$intent_id = $order->get_transaction_id();
+
+			$request = Get_Intention::create( $intent_id );
+			$intent  = $request->send( 'wcpay_get_intent_request', $order );
+
 			$payment_type = $this->is_payment_recurring( $order->get_id() ) ? Payment_Type::RECURRING() : Payment_Type::SINGLE();
 
 			$metadata_from_intent = $intent->get_metadata(); // mobile app may have set metadata.
 			$metadata_from_order  = $this->get_metadata_from_order( $order, $payment_type );
 			$merged_metadata      = array_merge( (array) $metadata_from_order, (array) $metadata_from_intent ); // prioritize metadata from mobile app.
 
-			$this->payments_api_client->prepare_intention_for_capture(
-				$intent_id,
-				$merged_metadata
-			);
+			$wcpay_request = Update_Intention::create( $intent_id );
```

Comment:
I'm also not completely inline with the name here. Considering `capture_charge` includes plenty of requests, let's have all of them assigned to a distinctive variable, ex. `$get_request` and `$update_request` instead of `$request` and `$wcpay_request`.

Reply:
I honestly don't like the idea of creating a new class and change one line for the sake of covering something that is hard to sport... I mean, where is the line? Shall we create a new class, require it manually (because we don't have autoloader) and write tests for that class? We are already covering scenarios like these and famous `receipt_email` parameter is by default set to correct value. 

If we decide to use this approach, we are setting examples that it is fine to create a bunch of classes to mimic specific use case and after some time you will have a bunch of them and good luck navigating them and onboard someone new into this where you have multiple classes for same requests where only one parameter is changed.

Reply:
It makes sense to include the empty parameter with the defaults of `Update_Intention`, the way it is right now.

However, I had not noticed the [default value](https://github.com/Automattic/woocommerce-payments/blob/b255f2d9cc6f23bc6b89a0c85880bc069959f545/includes/core/server/request/class-update-intention.php#L24) until now, even though it's pretty important. Let's add a comment there, and I think we can close this discussion.